server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -657,6 +658,13 @@ private Query getWildcardQuerySingle(String field, String termStr) throws ParseE
 
     @Override
     protected Query getRegexpQuery(String field, String termStr) throws ParseException {
+        final int maxAllowedRegexLength = context.getIndexSettings().getMaxRegexLength();
+        if (termStr.length() > maxAllowedRegexLength) {
+            throw new IllegalArgumentException(
```

Comment:
Maybe this could be a ParseException, that way we would catch and wrap it in QueryStringQueryBuilder#doToQuery into a QueryShardException, which might make the failure more consistent. On the other hand that might as well have disadvantages I didn't think about. Just a suggestion.

Reply:
I think it's consistent with the way we apply the other limits and a ParseException would be misleading ?

Reply:
Sound like a good reason, I'm fine with either way.